inorbit_republisher/inorbit_republisher/republisher.py

```diff
@@ -153,7 +154,14 @@ def callback(msg, repub=repub):
                         node.get_logger().warning(f"Failed to serialize message: {e}")
 
                 if val is not None:
-                    pubs[topic].publish(String(data=f"{key}={val}"))
+                    if "on_change" in repub and repub["on_change"]:
+                        # Publish only when value changes
+                        data = String(data=f"{key}={val}")
+                        if key not in last_data or last_data[key] != data:
+                            last_data[key] = data
+                            pubs[topic].publish(data)
+                    else:
+                        pubs[topic].publish(String(data=f"{key}={val}"))
```

Comment:
I think we can avoid some repetition reorganizing the code a little bit. Also I think that the key used in `last_data` must contain the topic to prevent two republishes on different topics configured with `on_change` from interfering with each other.

```suggestion
                    if repub.get("on_change"):
                        uniq_key = f"{topic}:{key}"
                        if last_data.get(uniq_key) == str(val):
                            return
                        last_data[uniq_key] = str(val)
                    pubs[topic].publish(String(data=f"{key}={val}"))
```

Reply:
true, thanks for the suggestion, that's more readable :+1: